### PR TITLE
dotCMS/core#23460 Container when delete a content type tab it should focus on another

### PR DESCRIPTION
**Pull request by zulqarnainvd**
### Proposed Changes
* show empty content type message when the content type is empty.
* focus on the editor when clicking on the tab.
* focus on another tab and code when the content type removes and the content type if exists.

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts [hunk 1]
```diff
@@ -140,7 +182,10 @@ export class DotContentEditorComponent implements OnInit {
                             })
                         );
 
-                        this.updateActiveTabIndex(this.getcontainerStructures.length);
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:143]: did you try doing this on `ngAfterViewInit()` ?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts [hunk 2]
```diff
@@ -75,9 +77,49 @@ export class DotContentEditorComponent implements OnInit {
      * @param close - This is the function that closes the modal.
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:77]: remove

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts (hunk 1 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:143]: it doesn't work in this case.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts [hunk 3]
```diff
@@ -75,9 +77,49 @@ export class DotContentEditorComponent implements OnInit {
...
+        this.updateActiveTabIndex(currentTabIndex);
+        this.foucsCurrentEditor(currentTabIndex);
+    }
+
+    /**
+     * Focus current editor
+     * @param {number} tabIdx
+     * @memberof DotContentEditorComponent
+     */
+    foucsCurrentEditor(tabIdx: number) {
```
[the remark this answers is not in the document]
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts]: updated

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts (hunk 2 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts]: updated

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts [hunk 4]
```diff
@@ -64,20 +65,60 @@ export class DotContentEditorComponent implements OnInit {
...
+        if (currentIndex > 0) {
+            // do nothing
+        } else if (this.getcontainerStructures.length > 0) {
+            currentIndex = currentIndex + 1;
+        } else {
+            // empty tab index
+            currentIndex = 0;
+        }
+
+        return currentIndex;
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:121]:
```suggestion
        return index > 0 ? index-1 : this.getcontainerStructures.length > 0 ? index : 0
```

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:121]: Yeah, empty if not a good practica.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts (hunk 1 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:143]: What about: https://stackoverflow.com/questions/38709923/why-is-requestanimationframe-better-than-setinterval-or-settimeout maybe?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts (hunk 4 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts]: updated

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts (hunk 1 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts:153]: updated

**Review by hmoreras — APPROVED** [review]

**Pull request merged**